File: src/main/java/gr/apt/persistence/dto/PersonBasicInfoDto.java
package gr.apt.persistence.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import gr.apt.persistence.enumeration.Job;
import lombok.Data;

import java.math.BigInteger;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class PersonBasicInfoDto {
    private BigInteger id;
    private String fname;
    private String lname;
    private String email;
    private String username;
    private Job job;
}

File: src/main/java/gr/apt/dto/person/PersonBasicInfoDto.kt
package gr.apt.dto.person

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import gr.apt.persistence.enumeration.Job
import java.math.BigInteger

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class PersonBasicInfoDto(
    var id: BigInteger? = null,
    var fname: String? = null,
    var lname: String? = null,
    var email: String? = null,
    var username: String? = null,
    var job: Job? = null
)